optimizely/config/polling_manager.go

```diff
@@ -53,42 +53,42 @@ type PollingProjectConfigManager struct {
 	exeCtx utils.ExecutionCtx // context used for execution control
 }
 
-func (cm *PollingProjectConfigManager) activate(initialPayload []byte, init bool) {
-
-	update := func() {
-		var e error
-		var code int
-		var payload []byte
-		if init && len(initialPayload) > 0 {
-			payload = initialPayload
-		} else {
-			payload, code, e = cm.requester.Get()
-
-			if e != nil {
-				cmLogger.Error(fmt.Sprintf("request returned with http code=%d", code), e)
-			}
-		}
-
-		projectConfig, err := datafileprojectconfig.NewDatafileProjectConfig(payload)
-		if err != nil {
-			cmLogger.Error("failed to create project config", err)
+// SyncConfig gets current datafile and updates projectConfig
+func (cm *PollingProjectConfigManager) SyncConfig(datafile []byte) {
+	var e error
+	var code int
+	if len(datafile) == 0 {
```

Comment:
Can we do a nil check and pass in `nil` from the ticker sync call below? It is more accurate to treat it as nil than an empty byte array

Reply:
we will be able to pass both, so we need to check for len of datafile when it's not nil, and also for nil itself

Reply:
I am going by @pawels-optimizely 's comment and leaving this as is.